### ISSUE TITLE
[GAME BUG] Dragon Force II Save issue

### ISSUE THREAD

**Issue by whitezombie2000** [AppCakeLtd/fenrir-bug-reports#2]
**Game Name**
Dragon Force II

**Game Region**
JP

**Game Source**
- [ ] Redump (cue/bin)
- [ ] Trurip (ccd/img)
- [ ] Other (iso)
- [x] Other (cue/bin) - Translation-patched Image
- [x] Other (ccd/img) - Translation-patched Image

**Describe the bug**
When continuing a game from the title screen in Dragon Force II, or accessing the save menu from in the game you get a black screen for about 30 seconds, that has been patched to English, and converted to ccd/img. The save menu does come up after 30 seconds. It should only be a few seconds for this menu to open. I can repeat this process with no cartridge inserted accessing the internal RAM, with the official backup RAM cart inserted, Saturn Gamer's Cartridge, and with the all in one Psuedo Kai cart inserted. I am using an unpatched BIN+CUE, that is patched with the Dragon_Force_II_English_v1006.xdelta patch following the read me instructions, then converted to ccd/img, and that is when I see the issue. If I create a patched BIN+CUE and play that way the save menu goes black for 20-25 seconds, some garbage is produced on screen, and it goes black. Using an unpatched BIN+CUE leaving the game untouched in its native Japanese language the save menu works fine and opens in a few seconds. I don't have any issue with my Phoebe and Rhea accessing the save menu in Dragon Force II patched and converted to ccd/img, played through it twice.

After the game is patched following the readme you will have a Track.ISO file. You can mount that file with Daemon Tools LITE and use ImgBurn to create your patched BIN+CUE file. You can play in this format but the game will lock at a black screen when you go to continue at the title screen.

Here's how I create my ccd images for Phoebe and Rhea.

Install Daemon Tools LITE (latest version is fine, no need to find an older version, and no need for DiscJuggler) and download a small program called CD Manipulator. I use version 2.70 Fix-7. Mount the patched Dragon Force II BIN+CUE with Daemon Tools LITE, open CD Manipulator, select the upper left icon, Read from CD to image file, change the drop down box to Multi-Session mode, tick Extended Settings, tick Analyze Sub Code, click browse to save the file somewhere, and click OK. This will create a .cdm file, .cue file, .img file, and .sub file. Rename the .cdm file to .ccd, then copy the .ccd, .img, and .sub file to the SD card game name folder. When in this format the save menu can be accessed but it takes about 30 seconds to open.

**Console Revision**
- [x] VA0
- [ ] VA0.5
- [ ] VA0.8
- [ ] VA1

**Console Region**
- [ ] PAL
- [x] NTSC-U
- [ ] NTSC-J
- [ ] Other

**Firmware and Loader Version**
- Firmware Version: da719d1b
- Loader Version: 951db30

**Cartridge Attached**
(Please select more than one if it applies)
- [x] None
- [x] Saturn Gamer's Cartridge
- [x] Pseudo Saturn Kai / Kai Lite - Red Cartridge all in one
- [ ] Pseudo Saturn Kai - Custom Made
- [x] Original Save Cartridge
- [ ] Third party Save Cartridge
- [ ] Original 1MB RAM Expansion
- [ ] Original 4MB RAM Expansion
- [ ] MPEG Card

**To Reproduce**
Steps to reproduce the behaviour:
1. Launch Game and go to continue at the title screen or try to save in game.

**Flex Cable Attached**
- [x] Original Flex Cable
- [x] Replacement Flex Cable provided by 8BitMods
- [ ] Replacement Flex Cable Sourced by myself (please state where from and part number)

**Expected behaviour**
Go to continue at the game title screen and access game saves within  a few seconds.

**Comment by privateye** [comment]
I have not been able to reproduce these issues in the English translated version of Dragon Force 2. For me, the load/save menus appear almost instantaneously This was my configuration at the time of testing:

**Game Source**
- [x] Redump (cue/bin)
- [ ] Trurip (ccd/img)
- [ ] Other (iso)
- [ ] Other (cue/bin)
- [ ] Other (ccd/img)

**Console Revision**
- [ ] VA0
- [x] VA0.5
- [ ] VA0.8
- [ ] VA1

**Console Region**
- [ ] PAL
- [ ] NTSC-U
- [x] NTSC-J
- [ ] Other

**Firmware and Loader Version**
- Firmware Version: da719d1b
- Loader Version: 951db30

**Cartridge Attached**
(Please select more than one if it applies)
- [ ] None
- [ ] Action Replay 3 in 1
- [x] Pseudo Saturn Kai / Kai Lite - Red Cartridge all in one
- [] Pseudo Saturn Kai - Custom Made
- [ ] Original Save Cartridge
- [ ] Third party Save Cartridge
- [ ] Original 1MB RAM Expansion
- [ ] Original 4MB RAM Expansion
- [ ] MPEG Card

**Flex Cable Attached**
- [x] Original Flex Cable
- [ ] Replacement Flex Cable provided by 8BitMods
- [ ] Replacement Flex Cable Sourced by myself (please state where from and part number)

**Additional context**
Games installed on SanDisk Ultra 128 GB microSDXC card, formatted with Panasonic SD Card Formatter 5.0.1.
Image created by converting an existing image that I had in CDI format to CloneCD format, using CD Manipulator with these settings: https://imgur.com/a/guVvuTf

**Comment by whitezombie2000** [comment]
Here are the two images I created. One patched BIN+CUE which goes to a black screen and a the other is .ccd which takes 30 seconds to load the save screen.

https://www.mediafire.com/file/mnihrjre0r5vt82/Dragon_Force_2.rar/file

https://www.mediafire.com/file/x38st43om4nenx5/DF2_patched_bin_cue.rar/file

I have played the .ccd on my Japanese console with a Phoebe and had no issue and I have played the .ccd on my NTSC console with Rhea and had no issue.

**Comment by privateye** [comment]
I have tried both images on my Fenrir and they exhibit the issues that you described. Rather than speculate why this is, the easiest solution is to download a working image which you can find at the link below. Once you have downloaded it, please let me know and I will remove the link.

https://www.dropbox.com/s/7gheifb0y596e6e/Dragon_Force_2_Fenrir_Compatible.7z?dl=1

**Comment by whitezombie2000** [comment]
Ok I figured out the issue after looking at your cue file. After creating the Track.iso with IsoBuster there is a small 4mb wave file created that is track 2 in Japanese. After I patched the ISO and recreated the BIN+CUE that is where it was going wrong. I was loading the patched Track.iso in Daemon Tools with no cue file and reripping with Imgburn. This was creating a BIN+CUE with out the wave file. The BIN created was about 4mb less than yours and the CUE looked like this after ripping with Imgburn.

FILE "DRAGON_FORCE_2.BIN" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00

I can tell you that I have burned this way to a real CD, played via Pseudo Saturn Kai, played on Rhea, and Phoebe with no issue. So it was playing the game without track 2 which is just a short audio file in Japanese.

So what I did is after patching the Track.iso I created a CUE file and then loaded that CUE into Daemon tools to recreate the BIN+CUE with Imgburn.

FILE "Track 01.iso" BINARY
  TRACK 01 MODE1/2048
    INDEX 01 00:00:00
	POSTGAP 00:02:00
FILE "Track 02.wav" WAVE
    TRACK 02 AUDIO
    PREGAP 00:02:00
    INDEX 01 00:00:00

After doing this I was getting the same size image file as the one you had. I recreated the .ccd, .img, and .sub file. the .img files were now the same size as yours and they matched 100% in a hex editor file compare. So after doing it this way it appears to work ok in BIN+CUE and .ccd format.
